File: DMOJ/decipher.cpp
#include <string>
#include <iostream>
#include <unordered_map>

using namespace std;
int main(){
	ios::sync_with_stdio(false);
	string input = "";
	cin >> input; 
	
	unordered_map<char, string> dict;
	
	dict['A'] = "-";
	dict['B'] = "--.";
	dict['C'] = ".--";
	dict['D'] = "-.-";
	dict['E'] = ".";
	dict['F'] = "..-";
	dict['G'] = "---.";
	dict['H'] = "..--";
	dict['I'] = "-.";
	dict['J'] = "---";
	dict['K'] = "...";
	dict['L'] = ".-.";
	dict['M'] = ".-.-";
	dict['N'] = ".-..";
	dict['O'] = "..";
	dict['P'] = ".--.";
	dict['Q'] = "--..";
	dict['R'] = ".-..-";
	dict['S'] = "....-";
	dict['T'] = "-..--";
	dict['U'] = "--";
	dict['V'] = "-.-..";
	dict['W'] = "--..-";
	dict['X'] = ".-.-.";
	dict['Y'] = ".....";
	dict['Z'] = "-----";
	
	int first_size = 0;

	
	while(first_size < input.length()){
		string output = "";

		for(int i=first_size; i<input.length(); i++){
			
		}
		
		first_size++;
	}
	
	
	return 0;
}

File: DMOJ/decipher.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Collections;
import java.util.Arrays;
import java.util.ArrayList;
import java.util.HashMap;
class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        ArrayList<String> output = new ArrayList<String>();
		HashMap<String, String> dict = new HashMap<String, String>();
		
		dict.put("A", "-");
		dict.put("B", "--.");
		dict.put("C", ".--");
		dict.put("D", "-.-");
		dict.put("E", ".");
		dict.put("F", "..-");
		dict.put("G", "---.");
		dict.put("H", "..--");
		dict.put("I", "-.");
		dict.put("J", "---");
		dict.put("K", "...");

		
		for(String input = br.readLine(); input != null; input = br.readLine()){
			int first_size = 0;
			int last_size = 0;
			while(first_size <= input.length()){ //change this
				StringBuilder sb = new StringBuilder(input);
				
			}
		}

        
    }
}
